src/memberPublicKey.ts

```diff
@@ -56,10 +57,17 @@ export async function deleteMemberPublicKey(
   publicKeyId: string,
   options: ServiceOptions,
 ): Promise<Result<undefined, MemberPublicKeyProblemType>> {
+  const memberBundleRequestModel = getModelForClass(MemberBundleRequestModelSchema, {
+    existingConnection: options.dbConnection,
+  });
   const memberPublicKey = getModelForClass(MemberPublicKeyModelSchema, {
     existingConnection: options.dbConnection,
   });
 
+  // Member bundle request should be deleted before the member public key
```

Comment:
This is just describing the code in its current form. It's not explaining why, which is the important part.

```suggestion
  // Defer the key deletion until the end to make retries possible, in case we fail to delete dependant records.
```